src/H5C.c

```diff
@@ -2364,8 +2766,22 @@ H5C_protect(H5F_t *f, const H5C_class_t *type, haddr_t addr, void *udata, unsign
 #ifdef H5_HAVE_PARALLEL
                                              coll_access,
 #endif /* H5_HAVE_PARALLEL */
-                                             type, addr, udata)))
+                                             type, addr, udata))) {
+            /* Print out meaningful message for VFD SWMR reader */
+            if (f->shared->vfd_swmr && !f->shared->vfd_swmr_writer) {
+                uint64_t tmp_tick_num = 0;
+
+                if (H5FD_vfd_swmr_get_tick_and_idx(f->shared->lf, TRUE, &tmp_tick_num, NULL, NULL) < 0)
```

Comment:
This breaks several layers of abstraction and should be convert to use the 'ctl' callback on the file, rather than directly calling into the SWMR VFD.